### PR TITLE
rework test for flexibility

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
These tests are about to be hit by changes to storages, such as sorting and moving slot numbers. The changes are noisy.
#### Summary of Changes
Make the noisy test change separately.
Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -5653,84 +5653,79 @@ pub mod tests {
...
-        );
-        accounts.add_root(SLOT);
+            AccountSharedData::new(128, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(129, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(256, 1, AccountSharedData::default().owner()),
+        ];
+
+        for i in 0..raw_expected.len() {
+            raw_expected[i].hash =
+                AccountsDb::hash_account(slot, &raw_accounts[i], &raw_expected[i].pubkey);
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:5709]: Just to confirm, do you have any concerns about replacing the hardcoded Hash values in the tests here?

File: runtime/src/accounts_db.rs [hunk 2]
```diff
@@ -5653,84 +5653,79 @@ pub mod tests {
...
-            )],
-        );
-        accounts.add_root(SLOT);
+            AccountSharedData::new(128, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(129, 1, AccountSharedData::default().owner()),
+            AccountSharedData::new(256, 1, AccountSharedData::default().owner()),
+        ];
+
+        for i in 0..raw_expected.len() {
+            raw_expected[i].hash =
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:5708]: Would this work too?
```rust
for x in &mut raw_expected {
    x.hash =
```

File: runtime/src/accounts_db.rs [hunk 3]
```diff
@@ -5653,84 +5653,79 @@ pub mod tests {
...
+                hash: Hash::default(),
                 lamports: 129,
-                slot: SLOT,
+                slot,
                 pubkey: pubkey128,
             },
             CalculateHashIntermediate {
                 version: 3,
-                hash: Hash::from_str("DpvwJcznzwULYh19Zu5CuAA4AT6WTBe4H6n15prATmqj").unwrap(),
+                hash: Hash::default(),
```
Comment (brooksprumo) [inline runtime/src/accounts_db.rs:5689]: Just to confirm, do you have any reservations about replacing the hardcoded hash values with computed values below?

File: runtime/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:5708]: I need the 'i' for the other loops. I'm sure there are other ways to do this, but this is just test code.

**Review by brooksprumo — COMMENTED** [review]
Looks good, just some simple questions I wanted to double check.

File: runtime/src/accounts_db.rs (hunk 3 above)
Comment (jeffwashington) [reply runtime/src/accounts_db.rs:5689]: no. I added those constants a few months ago. constants/hash values are a function of slot#.

File: runtime/src/accounts_db.rs (hunk 2 above)
Comment (brooksprumo) [reply runtime/src/accounts_db.rs:5708]: Fair, no worries.

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**